### ISSUE TITLE
CloudSQL Postgres Read Replica (errorReadReplicaInvalidOperation)

### ISSUE THREAD

**Issue by mr-andyd** [hashicorp/terraform-provider-google#7605]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v0.12.18
+ provider.google v3.41.0
+ provider.google-beta v3.41.0
+ provider.helm v0.10.5
+ provider.kubernetes v1.11.2
+ provider.random v2.2.1

### Affected Resource(s)

google_sql_database_instance

### Terraform Configuration Files

```tf
resource "google_sql_database_instance" "production-read-replica" {

  provider = google
  name = "database-replica"
  master_instance_name = "database-master"
  database_version = "POSTGRES_9_6"

  replica_configuration {
    connect_retry_interval = "30"
    failover_target        = false
  }

  settings {
    tier = "db-g1-small"

    database_flags {
      name  = "temp_file_limit"
      value = "2097152"
    }

    ip_configuration {
      ipv4_enabled    = "false"
      private_network = "projects/blah/global/networks/blah-blah"
    }
  }
}
```

### Expected Behavior

Terraform should apply and create the read replica.

### Actual Behavior

Error is produced - Error: Error, failed to create instance database-replica: googleapi: Error 400: The requested operation is not valid for a read-replica instance., errorReadReplicaInvalidOperation

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

- When setting the availability_type to ZONAL, it always tries to set it to REGIONAL when applying
- Even after importing the resource, it still tries to revert it to REGIONAL
- When applying Terraform, the resource is trying to set a maintenance window even though nothing is set within the resource

### References

* #0000

**Comment by edwardmedia** [comment]
@mr-andyd can you post the debug log?

**Comment by edwardmedia** [comment]
@mr-andyd is this still an issue?

**Comment by edwardmedia** [comment]
@mr-andyd is [this](https://github.com/hashicorp/terraform-provider-google/issues/7609) similar to what you see?

**Comment by mr-andyd** [comment]
@edwardmedia Apologies for not getting back sooner. Over the weekend I ended up upgrading Terraform from v0.12.18 to v0.13.5 and applied my module successfully. I am still unsure as to the reason why my replica was failing to create, however, I can confirm that I can no longer replicate the issue with the current Terraform version. I did nothing to change my modules other than upgrading my Terraform provider version.
